rasa/shared/core/domain.py

```diff
@@ -1774,7 +1763,7 @@ def slot_mapping_for_form(self, form_name: Text) -> Dict[Text, Any]:
         Returns:
             The slot mapping or an empty dictionary in case no mapping was found.
         """
-        return self.forms.get(form_name, {})
+        return self.forms.get(form_name, {})["required_slots"]
```

Comment:
How about making `required_slots` a constant? https://deviq.com/antipatterns/magic-strings

Reply:
The other changes are out of scope for this PR. let's try to keep the diff small.